[ARRISEOS-28289] Add notification on network process start

Jacek Manko <jacek.manko@redembedded.com>

diff --git a/Source/WebKit2/UIProcess/API/C/WKContext.h b/Source/WebKit2/UIProcess/API/C/WKContext.h
index 82334814..37b0e75d 100644
--- a/Source/WebKit2/UIProcess/API/C/WKContext.h
+++ b/Source/WebKit2/UIProcess/API/C/WKContext.h
@@ -51,6 +51,7 @@ typedef WKDataRef (*WKContextCopyWebCryptoMasterKeyCallback)(WKContextRef contex
 typedef void (*WKContextChildProcessDidCrashCallback)(WKContextRef context, const void *clientInfo);
 typedef WKContextChildProcessDidCrashCallback WKContextNetworkProcessDidCrashCallback;
 typedef WKContextChildProcessDidCrashCallback WKContextDatabaseProcessDidCrashCallback;
+typedef WKContextChildProcessDidCrashCallback WKContextNetworkProcessDidStartCallback;
 
 typedef struct WKContextClientBase {
     int                                                                 version;
@@ -91,6 +92,7 @@ typedef struct WKContextClientV2 {
 
     // Version 2.
     WKContextDatabaseProcessDidCrashCallback                            databaseProcessDidCrash;
+    WKContextNetworkProcessDidStartCallback                             networkProcessDidStart;
 } WKContextClientV2;
 
 // FIXME: Remove these once support for Mavericks has been dropped.
diff --git a/Source/WebKit2/UIProcess/WebContextClient.cpp b/Source/WebKit2/UIProcess/WebContextClient.cpp
index 64dbdefa..403f17c9 100644
--- a/Source/WebKit2/UIProcess/WebContextClient.cpp
+++ b/Source/WebKit2/UIProcess/WebContextClient.cpp
@@ -54,4 +54,12 @@ void WebContextClient::databaseProcessDidCrash(WebProcessPool* processPool)
     m_client.databaseProcessDidCrash(toAPI(processPool), m_client.base.clientInfo);
 }
 
+void WebContextClient::networkProcessDidStart(WebProcessPool* processPool)
+{
+    if (!m_client.networkProcessDidStart)
+        return;
+
+    m_client.networkProcessDidStart(toAPI(processPool), m_client.base.clientInfo);
+}
+
 } // namespace WebKit
diff --git a/Source/WebKit2/UIProcess/WebContextClient.h b/Source/WebKit2/UIProcess/WebContextClient.h
index b25c4d3b..9acc1d18 100644
--- a/Source/WebKit2/UIProcess/WebContextClient.h
+++ b/Source/WebKit2/UIProcess/WebContextClient.h
@@ -46,6 +46,7 @@ public:
     void plugInAutoStartOriginHashesChanged(WebProcessPool*);
     void networkProcessDidCrash(WebProcessPool*);
     void databaseProcessDidCrash(WebProcessPool*);
+    void networkProcessDidStart(WebProcessPool*);
 };
 
 } // namespace WebKit
diff --git a/Source/WebKit2/UIProcess/WebProcessPool.cpp b/Source/WebKit2/UIProcess/WebProcessPool.cpp
index 91cbd7a8..1dc8b650 100644
--- a/Source/WebKit2/UIProcess/WebProcessPool.cpp
+++ b/Source/WebKit2/UIProcess/WebProcessPool.cpp
@@ -478,6 +478,7 @@ NetworkProcessProxy& WebProcessPool::ensureNetworkProcess()
             process->reinstateNetworkProcessAssertionState(*m_networkProcess);
     }
 
+    m_client.networkProcessDidStart(this);
     return *m_networkProcess;
 }
 
